### PR TITLE
generate/remove serial number to markdown title

### PR DESCRIPTION
**Pull request by windwiny**

### CODE REVIEW

**Comment by sorbits** [comment]
I would prefer that new commands not be written to require `ruby18`.

Can you update this to work with ruby 2.0+?

File: Commands/Remove title number.plist [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+
+data = $stdin.read
+order = Order.new
+
+data2 = data.gsub(/^(#+)([^#\n]+)/) do |x|
+  hx, title = $1, $2.strip
+  next x if hx.size &gt; order.maxl
+
+  title.sub! /^(\d+\.)+\s+/, ''
```
Comment (sorbits) [inline Commands/Remove title number.plist]: I think this should be `/^[\d.]+\s+/ to support section numbers with and without trailing period (I personally do not add a trailing period when I enumerate sections)

Comment (windwiny) [inline Commands/Remove title number.plist]: this may be accidentally deleted title start with number and space

**Comment by sorbits** [comment]
Thanks for accommodating my request and sorry for taking so long to respond.

I have a few other things:

1. In _Remove title number_ you have this line:

        "#{hx}  #{title}"

    Is the double space intentional? If not, I would prefer to remove it, as Remove title numbers should be the reverse of generating title numbers, but right now, running the two actions in succession result in extra spaces.

2. In _Remove title number_ you have this line:

        next x if hx.size > order.maxl

    While it also exist in _Generate title numbers_ it’s more of an eyesore here, because you have an entire class definition just to test against `order.maxl`.

    I would suggest that we simply “hardcode” limit in the regexp like this:

        /^(#{,6})([^#\n]+)/

3. I would like to rename the items, normally we refer to `<hn>` tags as “heading tags” and we actually have a _Headings_ submenu in the Markdown bundle, so I would suggest we move the items to the submenu and call them _“Add Numbers”_ and _“Remove Numbers”_ (note here I also use Capitalized Words as that is the styleguide for menu items, and I use _Add_ instead of _Generate_ because I think for most users they would see this as adding numbers to their headings).

If you agree with these changes, I can perform them before merging, or you can push an updated commit.

I will btw squash all commits into one before merging.

**Comment by windwiny** [comment]
Thanks for respond,  i agree you suggest.

And you can do something change and squash them.

**Comment by sorbits** [comment]
Merged as 7a8ff34351f96d0a2c15b0f6fe1b0c7091fb08a2, thanks!

I removed the double-space when adding numbers, as I don’t think this is a common convention (I assume it’s from the “double space after period”).

If you think this is common, I suggest we introduce a variable such as `TM_MARKDOWN_HEADING_SEPARATOR` or similar, this could then default to `. ` and would also allow people to get rid of the trailing period after the number.

I removed the key equivalent because I don’t think this action will be invoked often enough to justify reserving a key for it. People can always assign it one locally, if they use it a lot, though using <kbd>⌃⌘T</kbd> + `an` + ↩ feels efficient to me.

Also added menu item separators to the _Headings_ submenu and placed the new items at the top.

**Pull request closed without merging**